### PR TITLE
Added WorkType entity to DB, backend and frontend

### PR DESCRIPTION
**Pull request by mariyamango**

### CODE REVIEW

File: backend/src/main/java/com/example/backend/service/WorkTypeService.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.example.backend.service;
```
Comment (UrsulaMontag) [inline backend/src/main/java/com/example/backend/service/WorkTypeService.java:1]: Your pr looks great. I have also seen that you have implemented integration tests. Just one question. Was it the intention to generate the integration test for the service class? Integration tests are very recoursenintentive with the @SpringBootTest annotation and are mostly generated for the controller to test the integration of the different dependencies in actual application environment. As the controllers usually do not contain any business logic, it is sufficient to write just a few tests. 
The service classes, on the other hand, usually contain the business logics that we want to test in their own functionality within the class. Here we want to consider and test all possible cases. If possible, without dependencies on other classes. We achieve this by not executing the service tests in the application environment (i.e. without @SpringBootTest annotation), but by mocking the dependencies.
I hope this explanation was helpful.

File: backend/src/test/java/com/example/backend/controller/CarHealthControllerTest.java [hunk 2]
```diff
@@ -1,12 +1,10 @@
 package com.example.backend.controller;
```
Comment (UrsulaMontag) [inline backend/src/test/java/com/example/backend/controller/CarHealthControllerTest.java:2]: comment above

**Review by UrsulaMontag — APPROVED** [review]

**Review by pavlo-bystrytskyi — APPROVED** [review]

File: backend/src/test/java/com/example/backend/controller/CarHealthControllerTest.java [hunk 3]
```diff
@@ -39,30 +32,39 @@ class CarHealthControllerTest {
...
+    @Autowired
+    private WorkTypeService workTypeService;
+
+    @Autowired
     private IdGeneratorService idGeneratorService;
 
     @Autowired
     private ObjectMapper objectMapper;
 
+    @DirtiesContext
```
Comment (UrsulaMontag) [inline backend/src/test/java/com/example/backend/controller/CarHealthControllerTest.java:56]: You can also put @DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
on top of the whole test. Then you don't have to repeat it for every single test.

File: backend/src/test/java/com/example/backend/controller/CarHealthControllerTest.java [hunk 4]
```diff
@@ -89,98 +91,91 @@ void getCarById_shouldReturnCar() throws Exception {
...
         //GIVEN
-        String carId = "1";
-        Mockito.doNothing().when(carHealthService).deleteCarById(carId);
+        String id = "123";
+        CarDto carDto = new CarDto(id, "Model S", 2023, "VIN123456", 10000);
+        carHealthService.createCar(carDto);
         //WHEN THEN
-        mockMvc.perform(delete("/api/cars/{id}", carId))
+        mockMvc.perform(delete("/api/cars/{id}", id))
                 .andExpect(status().isNoContent());
```
Comment (UrsulaMontag) [inline backend/src/test/java/com/example/backend/controller/CarHealthControllerTest.java:156]: Nice :D

File: backend/src/test/java/com/example/backend/service/CarHealthServiceTest.java [hunk 5]
```diff
@@ -16,101 +16,125 @@
...
+        //GIVEN
+        when(carHealthRepository.findById("1")).thenReturn(Optional.empty());
+        //WHEN THEN
+        assertThrows(NoSuchElementException.class, () -> carHealthService.deleteCarById("1"));
+        verify(carHealthRepository, times(1)).findById("1");
+        verify(carHealthRepository, times(0)).delete(any(Car.class));
     }
-}
\ No newline at end of file
+}
```
Comment (UrsulaMontag) [inline backend/src/test/java/com/example/backend/service/CarHealthServiceTest.java:140]: Yes, thats very nice.

**Review by UrsulaMontag — APPROVED** [review]
Hey Marija, now everything looks perfect.

**Pull request merged**